crates/virtio-queue/src/lib.rs

```diff
@@ -735,12 +734,14 @@ impl<M: GuestAddressSpace> QueueStateT<M> for QueueState<M> {
         self.max_size
     }
 
-    fn actual_size(&self) -> u16 {
-        min(self.size, self.max_size)
-    }
+    fn set_size(&mut self, size: u16) -> Result<(), Error> {
```

Comment:
Sorry for not being explicit enough in my previous comment :(. The way we were thinking to address the issue is to not return an error anymore in case of a misconfiguration, and just ignore the value and log an error, something like:
```rust
    fn set_size(&mut self, size: u16) {
        if size > self.max_size() || size == 0 || (size & (size - 1)) != 0 {
            error!("virtio queue with invalid size: {}", size);
        }
    }
```
It's not necessary to explicitly set the value to the default one, because the `Queue` constructor is already initializing the size and addresses to the default ones. I would though define constants for the default addresses like you already did [here](https://github.com/rust-vmm/vm-virtio/pull/112/files#diff-32f28d703ff40356a906276eb71cdad39cc03aeacaf631c1afc157fc2cd7d641R34), but one for each queue part:
const DEFAULT_DESC_TABLE_ADDR: u64 = 0x0;
const DEFAULT_AVAIL_RING_ADDR: u64 = 0x0;
...

With this approach, we won't need to define other errors, and the logic [here](https://github.com/rust-vmm/vm-virtio/pull/112/files#diff-32f28d703ff40356a906276eb71cdad39cc03aeacaf631c1afc157fc2cd7d641R43) won't be needed anymore.

We should also update the negative tests for these setters, i.e. try to set an invalid value, and check that the value of size/addr remains the default one.